tests/framework/microvm.py

```diff
@@ -996,11 +997,19 @@ def restore_from_snapshot(
         # Adjust things just in case
         self.kernel_file = Path(self.kernel_file)
 
+        iface_overrides = []
+        if rename_interfaces is not None:
+            iface_overrides = [
+                {"iface_id": k, "host_dev_name": v}
+                for k, v in rename_interfaces.items()
+            ]
+
         self.api.snapshot_load.put(
             mem_backend=mem_backend,
             snapshot_path=str(jailed_vmstate),
             enable_diff_snapshots=snapshot.is_diff,
             resume_vm=resume,
+            network_overrides=iface_overrides,
```

Comment:
To avoid sending `network_overrides` if it's empty do something like this.

```
kwargs = {}
if network_overrides:
    kwargs["network_overrides"] = network_overrides

self.api.snapshot_load.put(
    ...
    **kwargs,
)
```